src/server/scripts/Spells/spell_holiday.cpp

```diff
@@ -824,6 +824,60 @@ class spell_brewfest_barker_bunny : public SpellScriptLoader
         }
 };
 
+enum TorchSpells
+{
+    SPELL_TORCH_TOSSING_TRAINING                    = 45716,
+    SPELL_TORCH_TOSSING_PRACTICE                    = 46630,
+    SPELL_TORCH_TOSSING_TRAINING_SUCCESS_ALLIANCE   = 45719,
+    SPELL_TORCH_TOSSING_TRAINING_SUCCESS_HORDE      = 46651,
+    SPELL_BRAZIERS_HIT                              = 45724
+};
+
+// 45724 - Braziers Hit!
+class spell_midsummer_braziers_hit : public SpellScriptLoader
+{
+    public:
+        spell_midsummer_braziers_hit() : SpellScriptLoader("spell_midsummer_braziers_hit") { }
+
+        class spell_midsummer_braziers_hit_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_midsummer_braziers_hit_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                if (!sSpellMgr->GetSpellInfo(SPELL_BRAZIERS_HIT))
+                    return false;
+                return true;
+            }
+
+            void HandleEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                Player* player = GetTarget()->ToPlayer();
+                if (!player || !player->HasAura(SPELL_BRAZIERS_HIT))
+                    return;
+
+                if ((player->GetAura(SPELL_TORCH_TOSSING_TRAINING) && GetStackAmount() == 8) || (player->GetAura(SPELL_TORCH_TOSSING_PRACTICE) && GetStackAmount() == 20))
+                {
+                    if (player->GetTeam() == ALLIANCE)
+                       player->CastSpell(player, SPELL_TORCH_TOSSING_TRAINING_SUCCESS_ALLIANCE, true);
+                    else if (player->GetTeam() == HORDE)
+                        player->CastSpell(player, SPELL_TORCH_TOSSING_TRAINING_SUCCESS_HORDE, true);
+                    player->RemoveAura(SPELL_BRAZIERS_HIT);
+                }
+            }
+
+            void Register() override
+            {
+                AfterEffectApply += AuraEffectApplyFn(spell_midsummer_braziers_hit_AuraScript::HandleEffectApply, EFFECT_0, SPELL_AURA_DUMMY, AuraEffectHandleModes(AURA_EFFECT_HANDLE_REAL | AURA_EFFECT_HANDLE_REAPPLY));
```

Comment:
this mask is already defined as `AURA_EFFECT_HANDLE_REAL_OR_REAPPLY_MASK` but i think `AURA_EFFECT_HANDLE_REAPPLY` should be enough